crates/client-api/src/auth.rs

```diff
@@ -140,43 +138,16 @@ impl IntoResponse for AuthorizationRejection {
         // Sensible fallback if no auth header is present.
         const REQUIRED: (StatusCode, &str) = (StatusCode::UNAUTHORIZED, "Authorization required");
 
-        log::trace!("Authorization rejection: {:?}", self.reason);
+        log::trace!("Authorization rejection: {:?}", self);
 
-        match self.reason {
-            AuthorizationRejectionReason::Jwt(JwtErrorKind::InvalidSignature) => ROTATED.into_response(),
-            AuthorizationRejectionReason::Header(rejection) => match rejection.reason() {
-                TypedHeaderRejectionReason::Missing => REQUIRED.into_response(),
-                _ => rejection.into_response(),
-            },
-            _ => INVALID.into_response(),
+        match self {
+            AuthorizationRejection::Jwt(e) if *e.kind() == JwtErrorKind::InvalidSignature => ROTATED.into_response(),
+            AuthorizationRejection::Jwt(_) | AuthorizationRejection::Header(_) => INVALID.into_response(),
+            AuthorizationRejection::Required => REQUIRED.into_response(),
         }
     }
 }
 
-#[derive(Debug)]
-enum AuthorizationRejectionReason {
-    Jwt(JwtErrorKind),
-    Header(TypedHeaderRejection),
-    MalformedTokenQueryString,
-    CantDecodeAuthorizationToken,
```

Comment:
That wasn't an error that could ever actually occur; we basically encoded the token to base64 and then decoded it from base64 and if the decoding failed we returned that error. It could've just been an unwrap() instead of a variant.